README.md

```diff
@@ -1,16 +1,148 @@
 # OpenCensus Specs
 
-## General Design
-* For details about the library structured see [Namespace and Package](https://github.com/census-instrumentation/opencensus-specs/blob/master/NamespaceAndPackage.md)
+This is a high level design of the OpenCensus library, some of the API examples may be written 
+(linked) in C++, Java or Go but every language should translate/modify them based on language 
+specific patterns/idioms. Our goal is that all libraries have a consistent "look and feel".
 
-## Trace Design
-* Trace API is described [here](https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/README.md)
+This repository uses terminology (MUST, SHOULD, etc) from [RFC 2119][RFC2119].
 
-* Data model is defined [here](https://github.com/census-instrumentation/opencensus-proto/blob/master/trace/trace.proto)
+## Overview
+Today, distributed tracing systems and stats collection tend to use unique protocols and 
+specifications for propagating context and sending data to backend processing systems. This 
+is true amongst all the large vendors, and we aim to standardize these APIs and data models.
 
-## Tags Design
+The OpenCensus library refers to a collection of tools/APIs that are intended to be used primarily
+by cloud and distributed applications for monitoring and performance debugging. OpenCensus is 
+intended to be the first open-source application performance management (APM) library available 
+in all the main programming languages including C/C++, Java, Go, Ruby, PHP, Python, C#, Node.js,
+Objective-C and Erlang.
 
-## Stats Design
+## Ecosystem Design
 
-## Supported Encodings
+![alt text][EcosystemLayers]
+
+### Layers
+
+#### Service Exporters
+Each backend service MUST implement this API to export data to their services.
+
+#### OpenCensus Library
+This is what we design/describe in the current document.
+
+#### Manually instrumented Frameworks
+We are going to instrument some of the most popular frameworks for each language using the 
+OpenCensus library to allow users to get traces/stats when they use these frameworks.
+
+#### Tools for automatic instrumentation
+Some of the languages may support libraries for automatic instrumentation. Java for example can use
+byte-code manipulation (monkey patching) to provide an agent that automatically instruments a 
+binary. Note: not all the languages support this.
+
+#### Application
+This is customer's application/binary.
+
+## Library Design
+
+### Namespace and Package
+* For details about the library package names structure see [Namespace and Package][NamespaceAndPackage].
+
+### Components
+This section focuses on the important components that each OpenCensus library must have to 
+support all the functionalities.
+
+Here is a layering structure of the proposed OpenCensus library:
+
+![alt text][LibraryComponents]
+
+
+#### Generic Context
```

Comment:
Propagated Context (hits much harder.. .generic is a filler word)

Reply:
I changed it as just Context, if you care about calling "propagated" we can change it later.